keras_cv/models/stable_diffusion/text_encoder.py

```diff
@@ -18,47 +18,59 @@
 
 
 class TextEncoder(keras.Model):
-    def __init__(self, max_length, vocab_size=49408, name=None, download_weights=True):
-        tokens = keras.layers.Input(shape=(max_length,), dtype="int32", name="tokens")
+    def __init__(self, config, name=None, download_weights=True):
+        tokens = keras.layers.Input(shape=[config['max_length'],], dtype="int32", name="tokens")
         positions = keras.layers.Input(
-            shape=(max_length,), dtype="int32", name="positions"
+            shape=[config['max_length'],], dtype="int32", name="positions"
         )
-        x = CLIPEmbedding(vocab_size, 768, max_length)([tokens, positions])
-        for _ in range(12):
-            x = CLIPEncoderLayer()(x)
+        x = CLIPEmbedding(config['vocab_size'], config['embed_dim'], config['max_length'])([tokens, positions])
+        for _ in range(config['num_blocks']):
+            x = CLIPEncoderLayer(
+                embed_dim=config['embed_dim'], 
+                num_heads=config['num_heads'], 
+                use_q_gelu=config['use_q_gelu']
+                )(x)
         embedded = keras.layers.LayerNormalization(epsilon=1e-5)(x)
         super().__init__([tokens, positions], embedded, name=name)
 
         if download_weights:
-            text_encoder_weights_fpath = keras.utils.get_file(
-                origin="https://huggingface.co/fchollet/stable-diffusion/resolve/main/kcv_encoder.h5",
-                file_hash="4789e63e07c0e54d6a34a29b45ce81ece27060c499a709d556c7755b42bb0dc4",
-            )
+            if config['version']=='v1':
+                text_encoder_weights_fpath = keras.utils.get_file(
+                    origin="https://huggingface.co/fchollet/stable-diffusion/resolve/main/kcv_encoder.h5",
+                    file_hash="4789e63e07c0e54d6a34a29b45ce81ece27060c499a709d556c7755b42bb0dc4",
+                )
+            else:
+                text_encoder_weights_fpath = keras.utils.get_file(
+                    origin="https://huggingface.co/Jobayer/stable_diffusion_v2/resolve/main/text_encoder_v2_1.h5",
```

Comment:
Does OpenClip share the same model architecture?

Reply:
yeah it is open clip